datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -372,15 +371,16 @@ void flush() {
     if (producer != null) {
       try {
         producer.flush(_producerFlushTimeoutMs, TimeUnit.MILLISECONDS);
-      } catch (InterruptException | TimeoutException e) {
-        // The KafkaProducer object should not be reused on an interrupted flush
+      } catch (Exception e) {
```

Comment:
Nitpick. Only concern I have here is we are masking all exceptions to a generic exception. There could be exceptions which are unexpected and we do not want to handle them? https://www.infoworld.com/article/2073800/beware-the-dangers-of-generic-exceptions.html

Reply:
Discussed with Vish offline. Here we are just closing the producer and then throwing the exception as is. So the callers should get notified of all exceptions and react accordingly. In general though, yes it's bad practice to catch generic 'Exception'.